### PR TITLE
Update Azure quotas for additional node families

### PR DESCRIPTION
**Pull request by micheleRP**
## Description
This adds to Azure quota prereqs + updates the style of the prereqs to make it a bit easier to digest. 

Resolves https://redpandadata.atlassian.net/browse/DOC-538
Review deadline: Friday Oct 18

## Page previews
[Create a BYOC Cluster on Azure - Prereqs](https://deploy-preview-92--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/)

## Checks

- [ ] New feature
- [X] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Comment by ligfx** [comment]
Adding `Standard Dv3-series vCPUs` to the list and noting what each instance type is used for makes sense to me.

I'm not sure I understand the point of making the top-level items checkboxes (and it feels a bit busy).

I wonder if it might make sense to pull the quotas out into their own heading, similar to the SKU restrictions (or a combined section with them?). We'll eventually add more detail there when we release higher tiers anyways.

File: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc [hunk 1]
```diff
@@ -16,17 +16,20 @@ Confirm you have a minimum version of Redpanda `rpk` v24.1. See xref:reference:r
...
 In the https://login.microsoftonline.com/[Azure Portal^], confirm that the dedicated subscription you intend to use with Redpanda includes the following: 
 
-* Role: The Azure user must have the _Owner_ role in the subscription.
-
-* Resources: The subscription must be registered for the following resource providers. See the https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types[Microsoft documentation^]. 
+[%interactive]
+- [ ] **Role**: The Azure user has the _Owner_ role in the subscription.
+- [ ] **Quota**: The subscription has enough total regional vCPUs and vCPUs per instance family in the region where you will use Redpanda. See the https://learn.microsoft.com/en-us/azure/quotas/view-quotas[Microsoft documentation^].
++
+* Standard LASv3-series vCPUs:  24 (3 Redpanda broker nodes)
```
Comment (c4milo) [inline modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:23]: We now deploy Kafka Connect too. @chapmanc will know more.

Comment (chapmanc) [reply modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:23]: If the value is disabled we do also deploy a kafka-connect node pool. The defaults are [here](https://github.com/redpanda-data/cloudv2/blob/9db1238416d98f9d852a2df845f46589795f0d16/apps/controlplane-api/defaults/defaulter.go#L618-L632) or specified by the `tier` and `connectors_machine_type` in the install pack: https://github.com/redpanda-data/cloudv2/blob/9db1238416d98f9d852a2df845f46589795f0d16/install-pack/24.2.yml#L4892

Comment (micheleRP) [reply modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc]: Thanks @chapmanc: Does [this section](https://deploy-preview-92--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/#check-azure-quota) look correct?

**Comment by micheleRP** [comment]
> Adding `Standard Dv3-series vCPUs` to the list and noting what each instance type is used for makes sense to me.
> 
> I'm not sure I understand the point of making the top-level items checkboxes (and it feels a bit busy).
> 
> I wonder if it might make sense to pull the quotas out into their own heading, similar to the SKU restrictions (or a combined section with them?). We'll eventually add more detail there when we release higher tiers anyways.

Thanks @ligfx: I moved quota into its own section and removed the checkboxes. Please eyeball the [prereqs ](https://deploy-preview-92--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/#prerequisites)again and let me know if it's clear!

**Comment by ligfx** [comment]
> Thanks @ligfx: I moved quota into its own section and removed the checkboxes. Please eyeball the [prereqs ](https://deploy-preview-92--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/#prerequisites)again and let me know if it's clear!

LGTM!

**Review by ligfx — APPROVED** [review]

**Review by c4milo — APPROVED** [review]

File: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc [hunk 2]
```diff
@@ -40,20 +40,17 @@ If it is not registered, run:
 az provider register --namespace 'Microsoft.Compute'
 ```
 
-* Feature: The subscription must be registered for Microsoft.Compute/EncryptionAtHost. See the https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft documentation^].
+* **Feature**: The subscription must be registered for Microsoft.Compute/EncryptionAtHost. See the https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft documentation^].
```
Comment (asimms41) [inline modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:43]:
```suggestion
* **Feature**: The subscription must be registered for EncryptionAtHost/Microsoft.Compute. See the https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft documentation^].
Just suggested a change in order to match the order in the command below.
```

Comment (ligfx) [reply modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:43]: The correct name is `Microsoft.Compute/EncryptionAtHost`, which is what Azure calls it in error messages. The command below could be reordered if needed.

**Review by asimms41 — APPROVED** [review]
LGTM. Just one small suggestion.

File: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc (hunk 2 above)
Comment (micheleRP) [reply modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:43]: changed command to: 
`az feature register --namespace Microsoft.Compute --name EncryptionAtHost`

**Review by chapmanc — APPROVED** [review]

**Pull request merged**